packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -87,107 +43,257 @@ function addToMapGroup<Key, Value>(
   }
 }
 
-function describeLiteralType(type: ts.Type): string {
-  if (type.isStringLiteral()) {
-    return JSON.stringify(type.value);
-  }
+function isNodeInsideReturnType(node: TSESTree.TSUnionType): boolean {
```

Comment:
[Bug] This function doesn't account for parenthesized types. This code trips it up:

```ts
declare function fn2(): (never | 'foo') | 123;
```

More nesting in types would also break it.

https://deploy-preview-10744--typescript-eslint.netlify.app/play/#ts=5.9.2&fileType=.ts&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVNQEYAKASgC55UQA3EGeAH3gHJEccWAoUSWBFOmx4CAJjKVi1Og2ZsOLUk3iFRAZgDcQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1ukQBNYmvAIYtaRUrTIcmyfJXyxELVBkiJo0DtEjgwAXxB6gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false